### PR TITLE
systemd-analyze: Add svg scaling options

### PR DESCRIPTION
**Pull request by rajmohanrgithub**
+ Scale the x-axis of the resulting plot by a factor (default 1.0)
+ Add activation timestamps to each bar

### CODE REVIEW

**Comment by rajmohanrgithub** [comment]
Below test was failing
1415/1514 systemd:dist / check-help-systemd-analyze                                   OK               0.08s
==
hence ci-fails/needs-rework.
==
fixed as below:
changed (having 85 characters including " " at beginning, hence failing)
     --scale-svg=FACTOR      Stretch the x-axis of the plot by FACTOR (default: 1.0)
to (<80 charcaters , 77 characters for this change including " " at beginning)
     --scale-svg=FACTOR      Stretch x-axis of plot by FACTOR (default: 1.0)
fixed the failure for above test case.
--
changed (91 characters including " " at beginning. hence failing for above test case)
     --detailed              Add more details to SVG plot, e.g. show activation timestamps
to (76 characters)
     --detailed              Detail SVG plot,e.g.show activation timestamps
==
test case passes now after the above change.

File: src/analyze/analyze.c [hunk 1]
```diff
@@ -102,6 +102,8 @@ RuntimeScope arg_runtime_scope = RUNTIME_SCOPE_SYSTEM;
 RecursiveErrors arg_recursive_errors = _RECURSIVE_ERRORS_INVALID;
 bool arg_man = true;
 bool arg_generators = false;
+double svg_timescale = 1.0;
+bool detailed_svg = false;
```
Comment (bluca) [inline src/analyze/analyze.c]: arg_detailed_svg

File: src/analyze/analyze.c [hunk 2]
```diff
@@ -102,6 +102,8 @@ RuntimeScope arg_runtime_scope = RUNTIME_SCOPE_SYSTEM;
 RecursiveErrors arg_recursive_errors = _RECURSIVE_ERRORS_INVALID;
 bool arg_man = true;
 bool arg_generators = false;
+double svg_timescale = 1.0;
```
Comment (bluca) [inline src/analyze/analyze.c]: arg_svg_timescale

**Review by bluca — CHANGES_REQUESTED** [review]
Please document the new options in man/systemd-analyze.xml. Please add a test case (superficial to just call with the new options is fine just to exercise it under sanitizers, no need to do complex output checks) in test/units/TEST-65-ANALYZE.sh

File: src/analyze/analyze.c (hunk 1 above)
Comment (rajmohanrgithub) [reply src/analyze/analyze.c]: sure, will change it.

**Comment by rajmohanrgithub** [comment]
> Please document the new options in man/systemd-analyze.xml. Please add a test case (superficial to just call with the new options is fine just to exercise it under sanitizers, no need to do complex output checks) in test/units/TEST-65-ANALYZE.sh

kindly review as the changes related to above request is completed.

File: man/systemd-analyze.xml [hunk 3]
```diff
@@ -1601,6 +1601,24 @@ io.systemd.credential:vmm.notify_socket=vsock-stream:2:254570042
         <xi:include href="version-info.xml" xpointer="v255"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--scale-svg=<replaceable>FACTOR</replaceable></option></term>
+
+        <listitem><para>When used with the <command>plot</command> command, the x-axis of the plot
+        can be stretched by FACTOR (default: 1.0).</para>
+
+        <xi:include href="version-info.xml" xpointer="v256"/></listitem>
```
Comment (bluca) [inline man/systemd-analyze.xml]: 257 here and below

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
I forgot, please also add completion support to shell-completion/bash/systemd-analyze for the new arguments

**Comment by rajmohanrgithub** [comment]
> I forgot, please also add completion support to shell-completion/bash/systemd-analyze for the new arguments

added. please check. thanks

**Pull request merged**